fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java

```diff
@@ -2655,7 +2655,7 @@ Bundle createSearchBundle(List<Resource> resources, FHIRSearchContext searchCont
             List<QueryParameter> logicalIdReferenceSearchParameters = new ArrayList<>();
             for (QueryParameter queryParameter : searchContext.getSearchParameters()) {
                 if (!queryParameter.isReverseChained()) {
-                    if (queryParameter.isChained()) {
+                    if (queryParameter.isChained() && !queryParameter.isCanonical()) {
```

Comment:
What if it is isChained() and isCanonical()?

Reply:
What we're looking for here with chained parameters is whether or not the reference contained a "logical" version (i.e. `.../_history/xx`), so we can then issue a warning message that we don't act on versioned references . Since that type of version is not valid for canonical references, we don't need to check canonicals. However, I just realized we also don't need to do the check for logical id-only with canonical references, so I'll move the !queryParameter.isCanonical() check up to the line above:
```
if (!queryParameter.isReverseChained() &&  !queryParameter.isCanonical()) {
```

Reply:
Can you add that as a comment in the code?

Reply:
done